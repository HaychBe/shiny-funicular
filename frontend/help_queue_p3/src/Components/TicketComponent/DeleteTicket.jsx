import axios from "axios";
import React from "react";
import Avatar from "react-avatar";
import { Col, Row, Button, Card, Accordion } from "react-bootstrap";
import * as Icon from "react-bootstrap-icons";
const DeleteTicket = ({
  id,
  title,
  trainee,
  trainer,
  cohort,
  description,
  topic,
}) => {
const deleteTicket = (e) => {
    console.log("delete ticket info");
    axios.delete("/delete/" + id, {
      headers: {
        'Access-Control-Allow-Origin' : '*'
        }
    }).then(
      (response) => {
        console.log(response);
      },
      (error) => {
        console.log(error);
      }
    );
  };
}

  export default DeleteTicket;