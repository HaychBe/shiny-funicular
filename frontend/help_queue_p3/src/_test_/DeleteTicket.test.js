import React from 'react';
import renderer from 'react-test-renderer';
import DeleteTicket from  '../Components/TicketComponent/DeleteTicket';

it('DeleteTicket should match the snapshot every time it is rendered', () => {
    const id = 2;
    const title = "Jest ";
    const topic = "Jest Test ";
    const traineeName = "Trainee 1";
    const cohort = "CloudNative";
    const trainer = "Vinesh Ghela";
    const description = "Jest Test description";
    const time = "06/05/2021, 11:46:13";
    const tree = renderer.create(<DeleteTicket id={id} title={title} topic={topic} traineeName={traineeName} cohort={cohort} trainer={trainer} description={description} time={time}/>).toJSON();
    expect(tree).toMatchSnapshot();
});
