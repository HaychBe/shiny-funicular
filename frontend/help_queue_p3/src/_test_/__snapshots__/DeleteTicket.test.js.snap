// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeleteTicket should match the snapshot every time it is rendered 1`] = `
<button
  className="btn btn-danger"
  disabled={false}
  onClick={[Function]}
  type="button"
>
   Delete 
</button>
`;

exports[`renders correctly 1`] = `
<button
  className="btn btn-danger"
  disabled={false}
  onClick={[Function]}
  type="button"
>
   Delete 
</button>
`;
